JavascriptWebAppSample/README.md

```diff
@@ -0,0 +1,46 @@
+# Javascript Web App Sample
+
+For javascript web applications that want access to resources like VSTS REST API's, they will have to support an authentication flow for their users. The [Azure Active Directory Authentication Library (ADAL) for JS](https://github.com/AzureAD/azure-activedirectory-library-for-js) enables javascript application developers to setup inerative authentication flows and obtain access tokens for API usage.
+
+## Sample Application
+
+This buildable sample will walk you through the steps to create a single page javascript application which uses ADAL to authenticate a user via an interactive prompt and display all projects contained in a VSTS account/TFS collection.
+
+To run this sample you will need:
+* Http-server. You can download [NPM http server](https://www.npmjs.com/package/http-server) if you need one.
+* An Azure Active Directory (AAD) tenant. If you do not have one, follow these [steps to set up an AAD](https://docs.microsoft.com/en-us/azure/active-directory/develop/active-directory-howto-tenant)
+* A user account in your AAD tenant
+* A VSTS account backed by your AAD tenant where your user account has access. If you have an existing VSTS account not connected to your AAD tenant follow these [steps to connect your AAD tenant to your VSTS account](https://www.visualstudio.com/en-us/docs/setup-admin/team-services/manage-organization-access-for-your-account-vs)
+
+## Step 1: Clone or download vsts-auth-samples repository
+
+From a shell or command line: 
+```no-highlight
+git clone https://github.com/Microsoft/vsts-auth-samples.git
+```
+
+## Step 2: Register the sample application with you Azure Active Directory tenant
+
+1. Sign in to the [Azure Portal](https://portal.azure.com).
+2. On the top bar, click on your account and under the Directory list, choose the Active Directory tenant where you wish to register your application.
+3. On the left hand navigation menu, select `Azure Active Directory`.
+4. Click on `App registrations` and select `New application registration` from the top bar.
+5. Enter a `name` for you application, ex. "Adal JS sample", choose `Web app/API` for `application type`, and enter `http://localhost:8080` for the `Redirect URI`. Finally click `create` at the bottom of the screen.
+6. Save the `Application ID` from your new application registration. You will need it later in this sample.
+7. Grant Permissions for VSTS. Click `Required permissions` -> `add` -> `1 Select an API` -> type in and select `Microsoft Visual Studio Team Services` -> check the box for `Have full access to...` -> click `Save` -> click `Grant Permissions` -> click `Yes`.
+
```

Comment:
I can add a few, but there are a lot of steps and screen shot for each will make the document so long it may discourage users from completing the sample.